doc/SAI-Proposal-Prefix-Compression.md

```diff
@@ -190,22 +194,104 @@ typedef struct _sai_prefix_compression_api_t
 } sai_prefix_compression_api_t;
 ```
 
+## 2.2 ACL with Prefix Compression
+
+Enhanced support for meta-data field matching in ACL. New table attributes enable the configuration of source and destination prefix compression tables during the creation of an ACL table. Additionally, new field entry attributes allow matching on specific meta-data from either the source or destination prefix tables.
+
+
+## 2.2.1 ACL with Prefix Compression
+
+New table attributes enable the configuration of both source and destination prefix compression tables, which are utilized in field matching. The prefix compression SAI object ID is stored within the ACL table. An ACL table can be configured to use different prefix compression tables for the source and destination, or the same table for both. This flexibility extends to different ACL tables as well, allowing the same prefix compression table to be reused across multiple ACL tables or unique tables to be created for each ACL table.
+
+When performing meta-data entry lookup in an ACL, the prefix compression object ID is retrieved from the ACL table and dereferenced for a separate lookup in the prefix compression table to obtain the meta-data. Ideally, this lookup is performed in hardware and occurs simultaneously with the ACL table lookup.
+
+Example: Two prefix compression tables, Table_1 and Table_2, are created. ACL Table_1 is then configured with the attributes SAI_ACL_TABLE_ATTR_SRC_PREFIX_COMPRESSION_TABLE set to Prefix Compression Table_1 and SAI_ACL_TABLE_ATTR_DST_PREFIX_COMPRESSION_TABLE set to Prefix Compression Table_2. Under ACL Table_1, two entries are created. Entry_1 references meta-data from Prefix Compression Table_1, and Entry_2 references meta-data from Prefix Compression Table_2. During the pipeline process for Entry_1, the ID of Prefix Compression Table_1 is dereferenced from ACL Table_1, and a lookup is performed on Prefix Compression Table_1, which is a separate table in the pipeline. The result of this lookup determines whether Entry_1 hits or misses in the ACL. The same process is applied to Entry_2, but the lookup is performed in Prefix Compression Table_2.
+
```

Comment:
Just for my understanding.
In your example
ACL_Table1 points to SRC_PREFIX_COMPRESSION_TABLE
ACL_Table1->Entry has match on the result of SRC_PREFIX_COMPRESSION_TABLE as well

Now pipeline is something like this.
packet derives the associated compression table based on the ACL_Table1 configured. Note there is no match criteria to derive the src compression table i.e. it is not an action.
Subsequently lookup is performed on the src compression table and result is some "metadata".
Next "metadata is used as match criteria in ACL_Table1 and result may be miss or hit. Normal ACL actions of hit or miss will take effect.

Is this understanding correct?

Reply:
Yes that is correct. If you feel my wording is not clear enough, I can tweak that part to make it better.  Maybe the way you described it provides a better picture.

Reply:
Yes, if you can capture the language so that pipeline behavior is clear it will be very helpful later on.

I have one more comment. In any pipeline typically you don't derive entire table from table, it is usually an entry into another table based on some lookup criteria.

Having said that, I don't see this as an issue, since all you are doing is leveraging the stages of ACL tables/groups. In hardware truly speaking there is no such derivation from the ACL tables and there is an implicit pipeline behavior such that prefix compression table is always before the ACL table. This is for the reason that metadata from compression table is looked up in the ACL table.

Can you please confirm this understanding as well.

Reply:
Yes, I updated the docs:
1. Moved that section to a new section to make it clear
2. Used some of your verbiage to make the wording clear

I hope that is better but I don't mind giving it another try. I know this is something new and I want to make sure it is clear for future users

Reply:
Thanks, Can you please capture following in your own words as well
....
I have one more comment. In any pipeline typically you don't derive entire table from table, it is usually an entry into another table based on some lookup criteria.

Having said that, I don't see this as an issue, since all you are doing is leveraging the stages of ACL tables/groups. In hardware truly speaking there is no such derivation from the ACL tables and there is an implicit pipeline behavior such that prefix compression table is always before the ACL table. This is for the reason that metadata from compression table is looked up in the ACL table.

...

Reply:
@JaiOCP, not a problem at all. Again please free free to point out anything, this has to be clear for future users. Especially since this has not been done before. Thank you for your help with this

Reply:
Thanks. Approved the PR